src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -310,6 +310,11 @@ public override BoundNode VisitCollectionElementInitializer(BoundCollectionEleme
 
         public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpression node)
         {
+            if (_inExpressionLambda && node.Syntax.IsKind(SyntaxKind.ObjectCreationExpression) && ((ObjectCreationExpressionSyntax)node.Syntax).Type is null)
```

Comment:
>if (_inExpressionLambda && node.Syntax.IsKind(SyntaxKind.ObjectCreationExpression) && ((ObjectCreationExpressionSyntax)node.Syntax).Type is null) [](start = 12, length = 145)

If I remember previous feedback from Aleksey, it's better to check the bound information, rather than fall back to a syntax check.
Maybe we could check for the `ImplicitNew` conversion?